provisioner/terraform/testdata/generate.sh

```diff
@@ -23,6 +23,48 @@ generate() {
 	fi
 }
 
+minimize_diff() {
+	for f in *.tf*.json; do
+		declare -A deleted=()
+		declare -a sed_args=()
+		while read -r line; do
+			# Deleted line (previous value).
+			if [[ $line = -\ * ]]; then
+				key="${line#*\"}"
+				key="${key%%\"*}"
+				value="${line#*: }"
+				value="${value#*\"}"
+				value="${value%\"*}"
+				declare deleted["$key"]="$value"
+			# Added line (new value).
+			elif [[ $line = +\ * ]]; then
+				key="${line#*\"}"
+				key="${key%%\"*}"
+				value="${line#*: }"
+				value="${value#*\"}"
+				value="${value%\"*}"
+				# Matched key, restore the value.
+				if [[ -v deleted["$key"] ]]; then
+					sed_args+=(-e "s|${value}|${deleted["$key"]}|g")
+					unset deleted["$key"]
+				fi
+			fi
+			if [[ ${#sed_args[@]} -gt 0 ]]; then
+				# Handle macOS compat.
+				if sed -h 2>&1 | grep -q "\[-i extension\]"; then
+					sed -i '' "${sed_args[@]}" "$f"
+				else
+					sed -i'' "${sed_args[@]}" "$f"
+				fi
+			fi
+		done < <(
+			# Filter out known keys with autogenerated values.
+			git diff -- "$f" |
+				grep -E "\"(terraform_version|id|agent_id|resource_id|token|random|timestamp)\":"
+		)
+	done
+}
```

Comment:
I think this can be performed by `jq` using `walk` or similar?

```
jq 'walk(if type == "object" then with_entries(select(.key | test("(terraform_version|id|agent_id|resource_id|token|random|timestamp)") | not)) else . end)'
```

Reply:
I think that would give you the current value, but not the previous (or whether or not it changed) 🤔.